### ISSUE TITLE
Using --skip-unused-stages with `FROM ${arg}` fails with "error building image: could not parse reference"

### ISSUE THREAD

**Issue by psmit** [GoogleContainerTools/kaniko#1220]
I was testing out the code that got merged in https://github.com/GoogleContainerTools/kaniko/pull/1165 (created by @JordanGoasdoue and merged by @tejal29 ). I noticed that using variable source images fails when using `--skip-unused-stages=true`

**Actual behavior**
Kaniko exits with the error `error building image: could not parse reference: 2`

**Expected behavior**
Image builds successfully.

**To Reproduce**
Given the Dockerfile:

```ARG BUILD_IMG=base_build
FROM ubuntu:19.10 AS base

FROM base as base_build

FROM ${BUILD_IMG} AS build_broker
RUN mkdir -p /hi/target
RUN cp /etc/os-release /hi/target/release

FROM base AS deploy_broker
COPY --from=build_broker /hi/target/release /
RUN cat /release
```

- Running 
`docker run -v /work/test_kaniko_docker:/workspace gcr.io/kaniko-project/executor:ae9c9b281377ed5370d2962122656b304af3280c --no-push` builds the images and shows the ubuntu 19.10 release info.
- Running `docker run -v /work/test_kaniko_docker:/workspace gcr.io/kaniko-project/executor:ae9c9b281377ed5370d2962122656b304af3280c --no-push --build-arg BUILD_IMG=ubuntu:20.04` builds the images and shows the ubuntu 20.04 release info. 
- Running `docker run -v /work/test_kaniko_docker:/workspace gcr.io/kaniko-project/executor:ae9c9b281377ed5370d2962122656b304af3280c --no-push --skip-unused-stages=true` builds the image also fine (not that there are no builds to skip)
- Running 
`docker run -v /work/test_kaniko_docker:/workspace gcr.io/kaniko-project/executor:ae9c9b281377ed5370d2962122656b304af3280c --no-push --build-arg BUILD_IMG=ubuntu:20.04  --skip-unused-stages=true` fails with:

```
error building image: could not parse reference: 2
INFO[0000] Resolved base name ubuntu:19.10 to base      
INFO[0000] Resolved base name ubuntu:20.04 to build_broker 
INFO[0000] Resolved base name base to deploy_broker     
INFO[0000] Retrieving image manifest 2                  
```

**Additional Information**
 - Dockerfile: see above
 - Build Context: none besides Dockerfile
 - Kaniko Image:  gcr.io/kaniko-project/executor:ae9c9b281377ed5370d2962122656b304af3280c (Digest: sha256:f6a248a290f0a6b88a46691322288bfe8d8543415ddbbf6aa2f6646abf86d36f )
 
 **Triage Notes for the Maintainers**
 

 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ No ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [ Yes ] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [ Yes ] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ Yes ] </li></ul>|

**Comment by tejal29** [comment]
@JordanGoasdoue Can you please verify this build on the given dockerfile. I am planning to do a release in a couple of hours and would love to get this in.

**Comment by JordanGoasdoue** [comment]
@tejal29 this is working now with the change done.
This is the exact same command used by @psmit with his Dockerfile : 
```
docker run -v /work/test_kaniko_docker:/workspace localhost:5000/executor-cross:latest --no-push --build-arg BUILD_IMG=ubuntu:20.04 --skip-unused-stages=true 
INFO[0000] Resolved base name ubuntu:19.10 to base      
INFO[0000] Resolved base name ubuntu:20.04 to build_broker 
INFO[0000] Resolved base name base to deploy_broker     
INFO[0000] Retrieving image manifest ubuntu:19.10       
INFO[0001] Retrieving image manifest ubuntu:19.10       
INFO[0003] Retrieving image manifest ubuntu:20.04       
INFO[0004] Retrieving image manifest ubuntu:20.04       
INFO[0005] Built cross stage deps: map[1:[/hi/target/release]] 
INFO[0005] Retrieving image manifest ubuntu:19.10       
INFO[0007] Retrieving image manifest ubuntu:19.10       
INFO[0008] Executing 0 build triggers                   
INFO[0008] Skipping unpacking as no commands require it. 
INFO[0008] Storing source image from stage 0 at path /kaniko/stages/0 
INFO[0013] Deleting filesystem...                       
INFO[0013] Retrieving image manifest ubuntu:20.04       
INFO[0014] Retrieving image manifest ubuntu:20.04       
INFO[0015] Executing 0 build triggers                   
INFO[0015] Unpacking rootfs as cmd RUN mkdir -p /hi/target requires it. 
INFO[0020] RUN mkdir -p /hi/target                      
INFO[0020] Taking snapshot of full filesystem...        
INFO[0020] Resolving 3260 paths                         
INFO[0021] cmd: /bin/sh                                 
INFO[0021] args: [-c mkdir -p /hi/target]               
INFO[0021] Running: [/bin/sh -c mkdir -p /hi/target]    
INFO[0021] Taking snapshot of full filesystem...        
INFO[0021] Resolving 3262 paths                         
INFO[0021] RUN cp /etc/os-release /hi/target/release    
INFO[0021] cmd: /bin/sh                                 
INFO[0021] args: [-c cp /etc/os-release /hi/target/release] 
INFO[0021] Running: [/bin/sh -c cp /etc/os-release /hi/target/release] 
INFO[0021] Taking snapshot of full filesystem...        
INFO[0021] Resolving 3263 paths                         
INFO[0022] Saving file hi/target/release for later use  
INFO[0022] Deleting filesystem...                       
INFO[0022] Base image from previous stage 0 found, using saved tar at path /kaniko/stages/0 
INFO[0022] Executing 0 build triggers                   
INFO[0022] Unpacking rootfs as cmd COPY --from=build_broker /hi/target/release / requires it. 
INFO[0024] COPY --from=build_broker /hi/target/release / 
INFO[0024] Resolving 1 paths                            
INFO[0024] Taking snapshot of files...                  
INFO[0024] RUN cat /release                             
INFO[0024] Taking snapshot of full filesystem...        
INFO[0024] Resolving 3218 paths                         
INFO[0025] cmd: /bin/sh                                 
INFO[0025] args: [-c cat /release]                      
INFO[0025] Running: [/bin/sh -c cat /release]           
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
INFO[0025] Taking snapshot of full filesystem...        
INFO[0025] Resolving 3218 paths                         
INFO[0025] No files were changed, appending empty layer to config. No layer added to image. 
INFO[0025] Skipping push to container registry due to --no-push flag
```

**Comment by tejal29** [comment]
Awesome. Thanks @JordanGoasdoue for a quick turnaround.
